App/src/main/java/com/gluonhq/chat/views/ChatPresenter.java

```diff
@@ -137,18 +136,10 @@ public void invalidated(Observable observable) {
 
     void updateMessages(Channel channel) {
         if (channel != null) {
-            channel.typing().addListener(new InvalidationListener() {
-                @Override
-                public void invalidated(Observable o) {
-                    if (channel.typing().get()) {
-                        channelName.setText(channel.displayName()+"...");
-                    } else {
-                        channelName.setText(channel.displayName());
-                    }
-                }
-            });
             createSortList(channel.getMessages());
-            channelName.setText(channel.displayName());
+            channelName.textProperty().bind(Bindings.createStringBinding(
+                    () -> channel.displayName() + (channel.isTyping() ? " ..." : ""),
```

Comment:
I'm ok with this for now, but in the future we might handle this with CSS as well, so that we have more flexibility (e.g. show an icon of a pen when someone is typing)